SpecRunner.html

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
```

Comment:
So Ed I am impressed with your code, specially if it is working as shown in your tests(getting the right score), your functions are robust and very few to calculate the whole thing, good job.
That being said I think you should really look into not creating literal code, you are doing it when you generate your html(element) environment, testing and even in the model. Lastly Ed the difference between the same challenge in Ruby and Javascript is that in Javascript using the specifics of this language you can control and update the view to accommodate changes to the score cards for previous frames based on the rolls just registering. I can not see that in your code, if it exits and I am missing it(my mistake) and if it does not you should really try and make use of this aspect. The HTML used to be fixed and presented as such while using Javascript would change it without reloading the whole state. Any way great code all together and great planning which I think are the results of great logic.

Reply:
Thanks Radu - I'm not sure I understand what you mean by creating literal code in my testing and model?

The html is very basic because I ran out of time, and figured something would be better than nothing in the browser (I basically emulated the html and interface from the thermostat project)! If I had more time I would make the page update selected elements on a roll by roll basis rather than (as at present) updating the whole thing.